sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2116,12 +2106,11 @@ void ProgramManager::addImages(sycl_device_binaries DeviceBinary) {
 }
 
 template <typename MultimapT, typename KeyT, typename ValT>
-void removeFromMultimapByVal(MultimapT &Map, const KeyT &Key, const ValT &Val,
-                             bool AssertContains = true) {
```

Comment:
Hi, @sergey-semenov 
We are working on removing all fallback assert related code in sycl-post-link and sycl runtime.
One thing to do is to clean up SYCL_ASSERT_USED meta data, I found you previous patch (https://github.com/intel/llvm/pull/19659/files#diff-78dd7f7ba0b6120dece1ae4ab5a09c9936ff654a1de2c31ff2dbb1fc58d90393R2123) added 'AssertContains' param in removeFromMultimapByVal function, could you help check whether we can remove it if we are going to remove assert metadata in sycl runtime? According to the code, I feel the "AssertContains" is not only used for 'assert', once any device image includes exported symbol, we need to keep them, if so, I suggest to rename the param to somthing like "ExpSymbolContains".
Thanks very much.

Reply:
Hi, @sergey-semenov 
Could you provide any insight here?
Thanks very much.